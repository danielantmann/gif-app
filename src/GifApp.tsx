import { CustomHeader } from "./shared/components/CustomHeader";
import { SearchBar } from "./shared/components/SearchBar";
import { PreviousSearches } from "./gifs/components/PreviousSearches";
import { GifList } from "./gifs/components/GifList";
import { useState } from "react";
import { getGifsByQuery } from "./gifs/actions/get-gifs-by-query";
import type { Gif } from "./gifs/interfaces/gif.interface";

export const GifApp = () => {
  const [previousTerms, setPreviousTerms] = useState<string[]>([]);
  const [gifs, setGif] = useState<Gif[]>([]);
  const handleTermClick = (term: string) => {
    console.log({ term });
  };

  const handleSearch = async (query: string = "") => {
    query.toLowerCase().trim();
    if (previousTerms.includes(query)) {
      return;
    }

    if (query.length === 0) {
      return;
    }

    const currentTerms = previousTerms.slice(0, 6);

    currentTerms.unshift(query);

    console.log(query);
    setPreviousTerms(currentTerms);

    const gifs = await getGifsByQuery(query);

    setGif(gifs);
  };
  return (
    <>
      <CustomHeader
        title="Buscador de Gif"
        description="Busca y comparte tus gifs"
      />
      <SearchBar
        placeholder="Buscar giffs"
        buttonTitle="Buscar"
        onQuery={handleSearch}
      />
      <PreviousSearches
        title="BÃºsqueda Previa"
        searches={previousTerms}
        onLabelClick={handleTermClick}
      />
      <GifList gifs={gifs} />
    </>
  );
};
