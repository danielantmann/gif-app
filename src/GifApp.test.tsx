import { describe, expect, test } from "vitest";
import { render } from "@testing-library/react";
import { GifApp } from "./GifApp";
describe("GifaApp", () => {
  test("should rendern components properly", () => {
    const { container } = render(<GifApp />);
    expect(container).toMatchSnapshot();
  });
});
