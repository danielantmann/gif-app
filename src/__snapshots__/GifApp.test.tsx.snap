// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifaApp > should rendern components properly 1`] = `
<div>
  <header
    class="content-center"
  >
    <h1>
      Buscador de Gif
    </h1>
    <p>
      Busca y comparte tus gifs
    </p>
  </header>
  <div
    class="search-container"
  >
    <input
      placeholder="Buscar giffs"
      type="text"
      value=""
    />
    <button>
      Buscar
    </button>
  </div>
  <div
    class="previous-searches"
  >
    <h2>
      BÃºsqueda Previa
    </h2>
    <ul
      class="previous-searches-list"
    />
  </div>
  <div
    class="gifs-container"
  />
</div>
`;
